src/unit-test-coverage/shared/src/coveragetest-idmap.c

```diff
@@ -1065,6 +1065,76 @@ void Test_OS_ObjectIdIterator(void)
     UtAssert_STUB_COUNT(OS_Unlock_Global_Impl, 2);
 }
 
+void Test_OS_ObjectIDInteger(void)
+{
+    /*
+     * Test Case For:
+     * OS_ObjectIdToInteger, OS_ObjectIdFromInteger, OS_ObjectIdEqual, OS_ObjectIdDefined
+     */
+    int32             actual;
+    OS_object_token_t token;
+    osal_id_t         typesI[OS_MAX_TOTAL_RECORDS];
+    osal_id_t         typesJ[OS_MAX_TOTAL_RECORDS];
+    uint32            intID;
+    int32             recordscount = 0;
+    osal_objtype_t    idtype;
+    char              str[OS_MAX_TOTAL_RECORDS];
+
+    for (idtype = 0; idtype < OS_OBJECT_TYPE_USER; ++idtype)
+    {
+        actual = OS_SUCCESS;
+        do
+        {
+            snprintf(str, sizeof(str), "%d", recordscount);
+
+            actual = OS_ObjectIdAllocateNew(idtype, str, &token);
+
+            if (actual == OS_SUCCESS)
+            {
+                typesI[recordscount] = token.obj_id;
+                intID                = OS_ObjectIdToInteger(typesI[recordscount]);
+                typesJ[recordscount] = OS_ObjectIdFromInteger(intID);
+
+                recordscount++;
+            }
+
+        } while (actual == OS_SUCCESS);
```

Comment:
The other thing I'd like to see as part of this loop is a check that `recordscount` does not exceed `OS_MAX_TOTAL_RECORDS`.  It shouldn't when things are working right.... but should confirm that.

Reply:
fixed